docs/development/journalist_api.rst

```diff
@@ -0,0 +1,414 @@
+              "is_read": false,
+              "size": 604,
+              "source_url": "/api/v1/sources/LBICF2DPGI3AMD74HIYXQ7QKQB3MCCKNMQ6Q4VPOL3OYWMIDMYDO6A37KOJCZNT3GVOEMCDHECM4S4OAXGGMZ452SD454A6EADXN3ZI%3D/",
+              "submission_id": 4,
+              "submission_url": "/api/v1/sources/LBICF2DPGI3AMD74HIYXQ7QKQB3MCCKNMQ6Q4VPOL3OYWMIDMYDO6A37KOJCZNT3GVOEMCDHECM4S4OAXGGMZ452SD454A6EADXN3ZI%3D/submissions/4/"
+          }
+      ]
+  }
+
+Get a single submission associated with a source [``GET``]
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires authentication.
+
+.. code:: sh
+
+  GET /api/v1/sources/<filesystem_id>/submissions/<int:submission_id>/
+
+Response 200 (application/json):
+
+.. code:: sh
+
+  {
+      "download_url": "/api/v1/sources/44YGZ2R7643TXBA65ZKRO5D6QH26RJ7NVDFMQJVFSMM6WA5W3ZDXNUYKGBTEUYGFCABBUEDLQ7OKS657WKOGUHFLVDLQ75GWTOX4D4A%3D/submissions/1/download/",
+      "filename": "1-olfactory_yuppie-msg.gpg",
+      "is_read": false,
+      "size": 604,
+      "source_url": "/api/v1/sources/44YGZ2R7643TXBA65ZKRO5D6QH26RJ7NVDFMQJVFSMM6WA5W3ZDXNUYKGBTEUYGFCABBUEDLQ7OKS657WKOGUHFLVDLQ75GWTOX4D4A%3D/",
+      "submission_id": 1,
+      "submission_url": "/api/v1/sources/44YGZ2R7643TXBA65ZKRO5D6QH26RJ7NVDFMQJVFSMM6WA5W3ZDXNUYKGBTEUYGFCABBUEDLQ7OKS657WKOGUHFLVDLQ75GWTOX4D4A%3D/submissions/1/"
+  }
+
+Add a reply to a source [``POST``]
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires authentication. Clients are expected to encrypt replies prior to
+submission to the server. Replies should be encrypted to the public key of the
+source.
+
+.. code:: sh
+
+  POST /api/v1/sources/<filesystem_id>/reply
+
+with the reply in the request body:
+
+.. code:: sh
+
+  {
+   "reply": "-----BEGIN PGP MESSAGE-----[...]-----END PGP MESSAGE-----"
+  }
+
+Response 201 created (application/json):
+
+.. code:: sh
+
+  {
+    "message": "Your reply has been stored"
+  }
+
+Replies that do not contain a GPG encrypted message will be rejected:
+
+Response 412 (application/json):
+
+.. code:: sh
+
+  {
+      "message": "You must encrypt replies client side"
+  }
+
+Delete a submission [``DELETE``]
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires authentication.
+
+.. code:: sh
+
+  DELETE /api/v1/sources/<filesystem_id>/submissions/<int:submission_id>/
+
+Response 200:
+
+.. code:: sh
+
+  {
+    "message": "Submission deleted"
+  }
+
+Download a submission [``GET``]
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires authentication.
+
+.. code:: sh
+
+  GET /api/v1/sources/<filesystem_id>/submissions/<int:submission_id>/download
+
+Response 200 will have ``Content-Type: application/pgp-encrypted`` and is the
+content of the PGP encrypted submission.
+
+Delete a Source and all their associated submissions [``DELETE``]
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires authentication.
+
+.. code:: sh
+
+  DELETE /api/v1/sources/<filesystem_id>/submissions
```

Comment:
I think this should be `DELETE /api/v1/sources/<str:filesystem_id>` because I would expect a `DELETE /submissions` to only delete submissions.

Reply:
agreed - that makes a lot more sense, implemented in 6a28f2b (`Journalist API: Modify URI for delete collection endpoint`)